capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -177,6 +176,14 @@ public boolean putPageView(String restaurantName, String restaurantKey) {
             PreparedQuery results = dataStore.prepare(query);
             Entity resultsEntity = results.asSingleEntity();
             if (resultsEntity == null) {
+                // Query for the name
+                Query nameQuery = new Query("RestaurantInfo")
+                        .setFilter(new FilterPredicate(
+                                "restaurantKey", Query.FilterOperator.EQUAL, restaurantKey));
+                PreparedQuery nameResult = dataStore.prepare(nameQuery);
+                Entity restaurant = nameResult.asSingleEntity();
+                String restaurantName = restaurant.getProperty("name").toString();
+
                 //Create entity
```

Comment:
Nit: Unrelated to this change, but there should be a space after "//". Feel free to skip addressing this if you don't need to make any other changes.